lib/resolver.js

```
@@ -1,39 +1,49 @@
         this.strictMode = strictMode;
 
-        extensions.push({
-            name: 'allOf',
-            rules: [{
-                name: 'items',
-                params: {
-                    items: Joi.array().items(Joi.object()).required()
-                },
-                validate(params, value, state, options) {
-                    options = Hoek.applyToDefaults(options, { allowUnknown: true });
-
-                    for (const schema of params.items) {
-                        const { error } = schema.validate(value, options);
-                        if (error) {
-                            const details = error.details[0];
-                            return this.createOverrideError(details.type, details.context, { path : details.path }, options, details.message);
+        this.joi = Joi.extend(
+            {
+                type: 'object',
+                base: Joi.object(),
+                coerce: {
+                    from: 'string',
+                    method(value) {
+                        if (typeof value !== 'string' || (value[0] !== '{' && !/^\s*\{/.test(value))) {
+                            return;
                         }
+
+                        try {
+                            return { value: Bourne.parse(value) };
+                        } catch (ignoreErr) { }
                     }
-                    return value;
                 }
-            }]
-        });
-
-        this.joi = Joi.extend(extensions);
+            },
+            {
+                type: 'array',
+                base: Joi.array(),
+                coerce: {
+                    from: 'string',
+                    method(value) {
+                        if (typeof value !== 'string' || (value[0] !== '[' && !/^\s*\[/.test(value))) {
+                            return;
+                        }
+                        try {
+                            return { value: Bourne.parse(value) };
+                        } catch (ignoreErr) { }
```

What is the impact of ignoring this error?